apcd-cms/src/client/src/components/Extensions/EditExtensionModal/EditExtensionModal.tsx

```diff
@@ -0,0 +1,394 @@
+  ) => {
+    const { ext_id } = values;
+    const url = `administration/update-extension/${ext_id}/`;
+
+    try {
+      setShowSuccessMessage(false);
+      const response = await fetchUtil({
+        url,
+        method: 'PUT',
+        body: values,
+      });
+
+      if (onEditSuccess && response) {
+        onEditSuccess(response);
+        setUserFields([
+          { label: 'Applicable Data Period', value: values.applicable_data_period || 'None' },
+          { label: 'Approved Expiration Date', value: values.approved_expiration_date || 'None' },
+          { label: 'Exception Status', value: values.ext_status },
+          { label: 'Exception Outcome', value: values.ext_outcome },
+          { label: 'Exception Notes', value: values.notes || 'None' },
+        ]);
+
+      }
+
+      setShowSuccessMessage(true);
+    } catch (error: any) {
+      if (error.response && error.response.data) {
+        // Use error message from the server response
+        setErrorMessage(
+          error.response.data.message ||
+            'An error occurred while saving the data. Please try again.'
+        );
+      } else {
+        // Use generic error message
+        setErrorMessage(
+          'An error occurred while saving the data. Please try again.'
+        );
+      }
+      setShowErrorMessage(true);
+    } finally {
+      actions.setSubmitting(false);
+    }
+  };
+
+  const formik = useFormik<FormValues>({
+    initialValues,
+    validationSchema,
+    onSubmit,
+    enableReinitialize: true,
+  });
+
+  const handleClose = () => {
+    setShowSuccessMessage(false);
+    setShowErrorMessage(false);
+  };
+/*
+  const userFields = [
+    {
+      label: 'Applicable Data Period',
+      value: extension.applicable_data_period
+        ? extension.applicable_data_period
+        : 'None',
+    },
+    {
+      label: 'Approved Expiration Date',
+      value: extension.approved_expiration_date
+        ? extension.approved_expiration_date
+        : 'None',
+    },
+    { label: 'Exception Status', value: extension.ex_status },
+    { label: 'Exception Outcome', value: extension.ex_outcome },
+    {
+      label: 'Exception Notes',
+      value: extension.notes ? extension.notes : 'None',
+    },
+  ];*/
+
+  return (
+    <>
+      <Modal
+        isOpen={isVisible}
+        onClose={onClose}
+        className={styles.customModal}
+        onClosed={handleClose}
+      >
+        <div className={`modal-header ${styles.modalHeader}`}>
+          <Label className={styles.customModalTitle}>
+            Edit Extension ID {extension.ext_id} for{' '}
+            {extension.org_name}
+          </Label>
+          <button
+            type="button"
+            className={`close ${styles.customCloseButton}`}
+            onClick={onClose}
+          >
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+        <ModalBody>
+          <Alert color="success" isOpen={showSuccessMessage}>
```

Comment:
Needs to be cleared and at some point so that if someone submits another change, you can tell that it went through and is not the same success message as the first submission. The user can't tell if this is a new message or not. 

Using the core components button, there is a isLoading prop that will spin when a submit button is clicked that would also be helpful for the user to know if the new submission is working.